### PR TITLE
[mieb] fixing lrap computation for multi-label classification

### PR DESCRIPTION
**Pull request by gowitheflow-1998**
Fixes https://github.com/embeddings-benchmark/mteb/issues/1792

Fixing the computation of the `lrap` metric for multi-label classification. `lrap` is supposed to [operate on continuous scores](https://scikit-learn.org/1.5/modules/generated/sklearn.metrics.label_ranking_average_precision_score.html), instead of predicted labels. This change largely smooths the large performance range in `lrap` across models brought by change in `samples_per_label`.

Was trying to fix the same for main but got a chain of failed tests (mainly because `main` is using `KNeighborsClassifier()` by itself - which I'm not sure would work in the expected way either; `mieb` is currently using `MultiOutputClassifer(estimator=LogisicRegression())`, which makes each label a separate binary classification problem).

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

### CODE REVIEW

**Review by isaac-chung — APPROVED** [review]
Great catch!
